geomfn: fix ST_Node panicking on MULTILINESTRING EMPTY

otan opened the pull request:
Release note (bug fix): Fixed a bug where ST_Node would panic if passed
in MULTILINESTRING EMPTY.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/60691)
<!-- Reviewable:end -->

otan committed:
geomfn: fix ST_Node panicking on MULTILINESTRING EMPTY

Release note (bug fix): Fixed a bug where ST_Node would panic if passed
in MULTILINESTRING EMPTY.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60691#-:-MTlMvsU29XPpqDZqilx:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geomfn/node.go, line 29 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/60691#-MTlMd8n8pFHGAO1PHb2:-MTlMd8n8pFHGAO1PHb3:bk4tbco) ([raw file](https://github.com/cockroachdb/cockroach/blob/acf0d3527cddcda067da4b1cd58e75528220c577/pkg/geo/geomfn/node.go#L29)):*
> ```Go
> 	}
> 
> 	// Return GEOMETRYCOLLECTION EMPTY if it is empty.
> ```

not a multi line string, based on the PostGIS docs? Or is this what PostGIS actually does?


<!-- Sent from Reviewable.io -->

otan commented:
___
*[pkg/geo/geomfn/node.go, line 29 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/60691#-MTlMd8n8pFHGAO1PHb2:-MTlP-ou8AmfcJXQ5L6z:b-9w1em8) ([raw file](https://github.com/cockroachdb/cockroach/blob/acf0d3527cddcda067da4b1cd58e75528220c577/pkg/geo/geomfn/node.go#L29)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

not a multi line string, based on the PostGIS docs? Or is this what PostGIS actually does?
</blockquote></details>

PostGIS returns GEOMETRYCOLLECTION EMPTY for MULTILINESTRING EMPTY, and panics for LINESTRING EMPTY.


<!-- Sent from Reviewable.io -->

otan commented:
bors r=sumeerbhola

thanks
